test/calc.js

```diff
@@ -196,4 +197,24 @@ describe('shouldUpdate', () => {
     const result = shouldUpdate(100, 110, propDefaults, state)
     expect(result.action).to.equal('unpin')
   })
+
+  it("should return an action of 'pin' if props.pin is set and not pinned, yet", () => {
+    const state = {
+      height: 100,
+      state: 'something',
+    }
+
+    const result = shouldUpdate(100, 110, { ...propDefaults, pin: true }, state)
+    expect(result.action).to.equal('pin')
+  })
+
+  it("should return an action of 'none' if props.pin is set and already pinned", () => {
+    const state = {
+      height: 100,
+      state: 'pined',
```

Comment:
There is a typo, should be `pinned` instead.

Reply:
Thanks @lunsdorf, I don't know if you'd be up for redoing this pr, I'm afraid I can't guarantee I'm going to merge it right away but as I said your PR provided a feature that seems to be in demand.

Reply:
Ok, see last comment :)